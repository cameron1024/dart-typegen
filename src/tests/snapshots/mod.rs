// #[test]
// fn test_foo() {
//     let dart = include_str!("./foo.dart");
//     let toml = include_str!("./foo.toml");
//
// }
