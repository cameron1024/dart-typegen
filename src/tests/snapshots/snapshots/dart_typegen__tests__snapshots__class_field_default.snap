---
source: src/tests/snapshots/mod.rs
expression: dart
---
final class Foo {
  final String x;
  final int y;

  const Foo({this.x = "hello", this.y = 123 + 234});

  static FooBuilder builder({String x = "hello", int y = 123 + 234}) =>
      FooBuilder(x: x, y: y);
  FooBuilder toBuilder() => FooBuilder(x: x, y: y);

  Map<String, dynamic> toJson() => {"x": x, "y": y};
  factory Foo.fromJson(Map<String, dynamic> json) => Foo(
    x: json["x"] == null ? "hello" : json["x"] as String,
    y: json["y"] == null ? 123 + 234 : json["y"] as int,
  );

  @override
  String toString() =>
      "Foo("
      "x: $x, "
      "y: $y"
      ")";
  @override
  bool operator ==(Object other) {
    if (identical(this, other)) {
      return true;
    }
    if (other is! Foo) {
      return false;
    }
    if (x != other.x) return false;
    if (y != other.y) return false;
    return true;
  }

  @override
  int get hashCode => Object.hashAll([x, y]);
}

/// Builder class for [Foo]
final class FooBuilder {
  String x;
  int y;

  FooBuilder({required this.x, required this.y});

  Foo build() => Foo(x: x, y: y);
}
